lib/Data/Chronicle/Writer.pm

```diff
@@ -141,7 +141,7 @@ sub set {
     die "Cannot store undefined values in Chronicle!" unless defined $value;
     die "You can only store hash-ref or array-ref in Chronicle!" unless (ref $value eq 'ARRAY' or ref $value eq 'HASH');
 
-    $value = JSON::to_json($value);
+    $value = encode_json($value);
```

Comment:
This will double-encode UTF-8 for database values, would need to encode JSON first, then send the UTF-8 version to Redis.